### DISCUSSION TITLE
[Dúvida] [Front] [Javascript] - Conexão com Banco de Dados

### DISCUSSION THREAD

**Discussion by thalita2000** [codaqui/institucional#97] [Q&A] [answered]
Boa tarde,
Qual a melhor forma de realizar a conexão do front end de uma página de login por exemplo para o banco de dados sql, sem a necessidade de um framework, pelo Javascript .

**Comment by endersonmenezes** [comment]
Eu não sou muito entendendor de Javascript, nem de front. Mas acho que isso cai em uma linha de boa prática, que o seu frontend não deveria ser responsável por essa conexão, mas pelo que vi aqui: https://stackoverflow.com/questions/857670/how-to-connect-to-sql-server-database-from-javascript-in-the-browser

Algo em torno disso iria funcionar, **expondo os seus dados de acesso**: 

```Javascript
<script>
    var objConnection = new ActiveXObject("adodb.connection");
    var strConn = "driver={sql server};server=QITBLRQIPL030;database=adventureworks;uid=sa;password=12345";
    objConnection.Open(strConn);
    var rs = new ActiveXObject("ADODB.Recordset");
    var strQuery = "SELECT * FROM  Person.Address";
    rs.Open(strQuery, objConnection);
    rs.MoveFirst();
    while (!rs.EOF) {
        document.write(rs.fields(0) + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;");
        document.write(rs.fields(1) + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;");
        document.write(rs.fields(2) + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;    ");
        document.write(rs.fields(3) + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;    ");
        document.write(rs.fields(4) + "<br/>");
        rs.movenext();
    }
</script>
```

**Comment by rioribeiro** [comment, marked as answer]
Já que o Enderson já respondeu sua pergunta, gostaria de reforçar a questão da boa prática porque venho me interessando bastante por cibersegurança nos últimos tempos. Primeiro, gostaria que você fizesse algumas perguntas a si mesma: por que seria interessante acessar um banco de dados pelo client side? É seguro manter uma conexão com algo tão valioso quanto um banco de dados diretamente no front end, onde o usuário pode interagir com mais liberdade e um simples F12 pode comprometer a segurança da sua aplicação? 

Agora é claro que, você pode estar apenas querendo se divertir com frontend + banco de dados, e não tem nada de errado nisso! Não necessariamente você vai aplicar isso num projeto público e/ou com dados sensíveis. De toda forma, se pressa for o motivo pelo qual você quer rapidamente conectar um banco de dados diretamente ao frontend, saiba os riscos que isso traria numa implementação "real".

Se tiver se interessada sobre o porque **não é** recomendável essa prática, leia: https://blog.betrybe.com/desenvolvimento-web/api-rest-tudo-sobre/

Bons estudos! 👋

**Reply by endersonmenezes** [reply]
Acho que foram perfeitas as suas colocações! Leiam isso @codaqui/membros e @thalita2000 !
